### PR TITLE
test: add type tests for within

### PR DESCRIPTION
**Pull request by timdeschryver**
**What**:

Follow up to #1034

**Why**:

I wanted to add generics to the within queries, but they already accept a generic.
I think adding a few tests could be helpful for potential changes in the future. 

**How**:

Added a few tests and saw that the tests were passing

**Checklist**:

- [ ] Documentation added to the
      [docs site](https://github.com/testing-library/testing-library-docs)
- [x] Tests
- [ ] TypeScript definitions updated
- [x] Ready to be merged
      

cc @nstepien

### CODE REVIEW

File: types/__tests__/type-tests.ts [hunk 1]
```diff
@@ -228,6 +229,18 @@ export async function testWaitFors() {
   await waitFor(async () => {})
 }
 
+export async function testWithin() {
+  const sut = within(document.body)
```
Comment (eps1lon) [inline types/__tests__/type-tests.ts:233]: What is "sut"?

Comment (timdeschryver) [reply types/__tests__/type-tests.ts:233]: It's an abbreviation for "System Under Test", I couldn't think of a better name.
I can change it if you want though.

Comment (eps1lon) [reply types/__tests__/type-tests.ts:233]: please 😄

Comment (timdeschryver) [reply types/__tests__/type-tests.ts]: No problem, renamed to `container`.

**Review by eps1lon — APPROVED** [review]

**Pull request merged**